src/ipred_prepare.rs

```diff
@@ -262,27 +250,23 @@ pub unsafe fn rav1d_prepare_intra_edges<BD: BitDepth>(
         .contains(Needs::TOP)
     {
         let sz_0 = tw << 2;
-        let top: *mut BD::Pixel = &mut *topleft_out.offset(1) as *mut BD::Pixel;
+        let top = &mut topleft_out[(topleft_origin + 1)..];
         if have_top {
             let px_have_1 = cmp::min(sz_0, w - x << 2);
-            memcpy(
-                top as *mut c_void,
-                dst_top as *const c_void,
-                (px_have_1 << 1) as usize,
-            );
+            for i in 0..px_have_1 {
+                top[i as usize] = *dst_top.offset(i as isize);
+            }
```

Comment:
We mistranslated this before.  It should be a `BD::pixel_copy`.  Could you also fix that?

Reply:
This should be fixed in a later commit